sdk/appconfiguration/app-configuration/src/secretReference.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { JsonSecretReferenceValue } from "./internal/jsonModels";
+import { ConfigurationSetting, ConfigurationSettingParam } from "./models";
+
+/**
+ * content-type for the secret reference.
+ */
+export const secretReferenceContentType =
+  "application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8";
+
+/**
+ * Necessary fields for updating or creating a new secret reference.
+ */
+export interface SecretReferenceValue {
+  /**
+   * Id for the secret reference.
+   */
+  secretId: string;
+}
+
+export const SecretReferenceHelper = {
+  /**
+   * Takes the ConfigurationSetting and returns the FeatureFlag.
+   */
+  fromConfigurationSetting: (
+    setting: ConfigurationSetting
+  ): ConfigurationSetting<SecretReferenceValue> => {
+    if (!isSecretReference(setting)) {
+      throw new Error("Not a SecretReference..");
+    }
+    let jsonSecretReferenceValue: JsonSecretReferenceValue;
+    try {
+      if (!setting.value || typeof setting.value !== "string") {
+        throw new Error("");
```

Comment:
Errors with no error messages aren't very friendly. Maybe this should be a TypeError since it looks like the user would have had to have given us an incorrect type of object here.

Reply:
I meant to add a message.. missed.. changed to Error now.
Will switch to TypeError/

Reply:
Let me know how that looks.